internal/internal_test.go

```diff
@@ -2497,3 +2501,1502 @@ func (p *testPluginError) Log(_ context.Context, event logs.EventV1) error {
 	p.events = append(p.events, event)
 	return fmt.Errorf("Bad Logger Error")
 }
+
+// mockExtProcStream is a mock implementation of the ProcessServer interface
+type mockExtProcStream struct {
+	ctx              context.Context
+	t                *testing.T
+	requestsToSend   []*ext_proc_v3.ProcessingRequest
+	receivedIndex    int
+	receivedMessages []*ext_proc_v3.ProcessingResponse
+	sendError        error
+	recvError        error
+}
+
+func (m *mockExtProcStream) Send(resp *ext_proc_v3.ProcessingResponse) error {
+	if m.sendError != nil {
+		return m.sendError
+	}
+	m.receivedMessages = append(m.receivedMessages, resp)
+	return nil
+}
+
+func (m *mockExtProcStream) Recv() (*ext_proc_v3.ProcessingRequest, error) {
+	if m.recvError != nil {
+		return nil, m.recvError
+	}
+
+	if m.receivedIndex >= len(m.requestsToSend) {
+		return nil, io.EOF
+	}
+
+	req := m.requestsToSend[m.receivedIndex]
+	m.receivedIndex++
+	return req, nil
+}
+
+// Additional methods required by the interface
+func (m *mockExtProcStream) SetHeader(metadata.MD) error {
+	return nil // No-op for tests
+}
+
+func (m *mockExtProcStream) SendHeader(metadata.MD) error {
+	return nil // No-op for tests
+}
+
+func (m *mockExtProcStream) SetTrailer(metadata.MD) {
+	// No-op for tests
+}
+
+func (m *mockExtProcStream) SendMsg(msg interface{}) error {
+	resp, ok := msg.(*ext_proc_v3.ProcessingResponse)
+	if !ok {
+		return fmt.Errorf("expected ProcessingResponse")
+	}
+	return m.Send(resp)
+}
+
+func (m *mockExtProcStream) RecvMsg(msg interface{}) error {
+	req, ok := msg.(*ext_proc_v3.ProcessingRequest)
+	if !ok {
+		return fmt.Errorf("expected ProcessingRequest")
+	}
+
+	received, err := m.Recv()
+	if err != nil {
+		return err
+	}
+
+	*req = *received
```

Comment:
this pointer deference and copy operation copies a mutex which is causing the linter to fail... do we need this mock implementation to actually copy the object?